### ISSUE TITLE
Performance issues when exporting many sheets with many rows

### ISSUE THREAD

**Issue by cervengoc** [JanKallman/EPPlus#492]
Hi guys,

I'm working on a project where we have to produce an Excel export having 15+ worksheets, each having about 10 columns and 30k rows. In this scenario the memory consumption of the current version is still very high and not acceptable for us.

What we tried is to save the export file sheet by sheet, this way we can avoid the unnecessary memory consumption, but on the other hand the saving and reloading of the package is very slow so it's not acceptable either.

I'd suggest an option to somehow "flush" a worksheet during export, so it would get written to the stream and would not hold the cells in memory any more.

If there is any other technique or suggestion for these kind of scenarios it would be great to hear about.

Thank you for all of your work.

**Comment by cervengoc** [comment]
Browsing the source code it looks like all worksheets get loaded eagerly when we first reference the `Workbook.Worksheets` collection. Another way to overcome my issue would be to introduce lazy loading of worksheets. So a worksheet would get loaded only when we first time reference `Cells` or anything similar which requires that the worksheet is loaded. 

In connection with this the `Worksheet.Save` method could be effective only when a worksheet had been loaded, otherwise it would do nothing.

I'm very little aware of the internals, so this all might not even make sense, please let me know in this case.

**Comment by cervengoc** [comment]
Digging deeper it seems more complex of course. However, as I think this feature would help many of us with large data sets on more sheets, I've started working on this, I'll put together a PR in the next days or so. The solution will not change the public API.

**Comment by cervengoc** [comment]
I've started the work and achieved some level of success, I can consecutively update an Excel file each sheet at a time. I wrote some tests. However, to be honest the sources are hard to read, lots of cross-references. And of course, the file is still read into memory streams. This idea and solution would make the most benefit if the file could be read and written in parts to the disk, not holding everything in memory. 

So for now I'll abandon this work, but anyone can check it out on my branch https://github.com/cervengoc/EPPlus/tree/feature/sheet-lazy-loading

If there are any plans to implement package handling on disk (not reading entire package into memory, and saving entire package), let me know and if I can find some time I'm happy to help.
